### PR TITLE
Better handling of attributes (boolean attributes + setAccuracy special case) for Wialon protocol

### PR DESCRIPTION
**Pull request by truppelito**
Pull request as discussed [here](https://www.traccar.org/forums/topic/motion-attribute-using-wialon-protocol/#post-81534).

### CODE REVIEW

File: src/main/java/org/traccar/protocol/WialonProtocolDecoder.java [hunk 1]
```diff
@@ -39,31 +39,31 @@ public WialonProtocolDecoder(Protocol protocol) {
     }
 
     private static final Pattern PATTERN_ANY = new PatternBuilder()
-            .expression("([^#]+)?")              // imei
-            .text("#")                           // start byte
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/WialonProtocolDecoder.java:43]: Please revert all these formatting changes.

File: src/main/java/org/traccar/protocol/WialonProtocolDecoder.java [hunk 2]
```diff
@@ -135,10 +135,41 @@ private Position decodePosition(Channel channel, SocketAddress remoteAddress, St
...
+                            position.setAccuracy(double_value);
+                        }
+                        else {
+                            position.set(key, double_value);
+                        }
                     } catch (NumberFormatException e) {
-                        position.set(paramParser.group(1).toLowerCase(), paramParser.group(2));
+                        if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("t")
+                            || value.equalsIgnoreCase("yes") || value.equalsIgnoreCase("y")
+                            || value.equalsIgnoreCase("on")) {
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/WialonProtocolDecoder.java:163]: Why do we need to check for all these values?

File: src/test/java/org/traccar/protocol/WialonProtocolDecoderTest.java [hunk 3]
```diff
@@ -75,6 +78,15 @@ public void testDecode() throws Exception {
...
 
+        verifyAttributes(decoder, text(
+                "#D#120319;112003;NA;NA;NA;NA;0.000;NA;NA;0;NA;NA;NA;NA;NA;motion:3:true"));
+
+        verifyAttributes(decoder, text(
+                "#D#120319;112003;NA;NA;NA;NA;0.000;NA;NA;0;NA;NA;NA;NA;NA;motion:3:false"));
+
+        Position p = (Position)decoder.decode(null, null, text(
+                "#D#120319;112003;NA;NA;NA;NA;0.000;NA;NA;0;NA;NA;NA;NA;NA;accuracy:2:12.3"));
+        assertEquals(p.getAccuracy(), 12.3, 0.001);
```
Comment (tananaev) [inline src/test/java/org/traccar/protocol/WialonProtocolDecoderTest.java:89]: Looks a bit messy. Maybe we can just skip this test.

File: src/test/java/org/traccar/protocol/WialonProtocolDecoderTest.java [hunk 4]
```diff
@@ -75,6 +78,15 @@ public void testDecode() throws Exception {
         verifyPositions(decoder, text(
                 "#B#110315;045857;5364.0167;N;06127.8262;E;0;155;965;7;2.40;4;0;14.77,0.02,3.6;AB45DF01145;"));
 
+        verifyAttributes(decoder, text(
+                "#D#120319;112003;NA;NA;NA;NA;0.000;NA;NA;0;NA;NA;NA;NA;NA;motion:3:true"));
+
+        verifyAttributes(decoder, text(
+                "#D#120319;112003;NA;NA;NA;NA;0.000;NA;NA;0;NA;NA;NA;NA;NA;motion:3:false"));
```
Comment (tananaev) [inline src/test/java/org/traccar/protocol/WialonProtocolDecoderTest.java:85]: One is probably enough. And use `verifyAttribute` so you can validate the value.

File: src/main/java/org/traccar/protocol/WialonProtocolDecoder.java (hunk 2 above)
Comment (truppelito) [reply src/main/java/org/traccar/protocol/WialonProtocolDecoder.java:163]: We don't. It's up to you to decide which values you'd like to check for.

Comment (tananaev) [reply src/main/java/org/traccar/protocol/WialonProtocolDecoder.java:163]: Maybe let's start with true/false for now.

**Comment by tananaev** [comment]
Merged, thank you.

**Comment by truppelito** [comment]
Thank you!

**Pull request merged**